cmd/copyEnumeratorInit.go

```diff
@@ -54,7 +54,7 @@ func (cca *cookedCopyCmdArgs) initEnumerator(jobPartOrder common.CopyJobPartOrde
 	jobPartOrder.DestLengthValidation = cca.CheckLength
 	jobPartOrder.S2SInvalidMetadataHandleOption = cca.s2sInvalidMetadataHandleOption
 
-	traverser, err = initResourceTraverser(cca.source, cca.fromTo.From(), &ctx, &srcCredInfo, &cca.followSymlinks, cca.listOfFilesChannel, cca.recursive, getRemoteProperties, cca.includeDirectoryStubs, func(common.EntityType) {}, cca.listOfVersionIDs)
+	traverser, err = initResourceTraverser(cca.source, cca.fromTo.From(), &ctx, &srcCredInfo, &cca.followSymlinks, cca.listOfFilesChannel, cca.recursive, getRemoteProperties, cca.includeDirectoryStubs, func(common.EntityType) {}, cca.listOfVersionIDs, cca.logVerbosity.ToPipelineLogLevel())
```

Comment:
I small suggestion:
I think we should consider creating an option bag similar to track2 here. The number of parameters are increasing and with more features like versioning and CPK coming to AzCopy, we cannot keep on updating the definition.
Also, while passing some attributes make sense while calling initResourceTraverser() 
with one operation (copy for example), passing those attributes with some other operation(list for example) may not even make sense (even passing a nil).

Alternatively, we can pass the whole `cooked` struct and get rid of changing function signature in one go.

What do you think?

Reply:
Ok let me follow up with a separate PR proposal.